### ISSUE TITLE
Windows shared lock with Python 3

### ISSUE THREAD

**Issue by barry-gallagher** [WoLpH/portalocker#64]
In portalocker.py for SHARED locks the code has a sys.major version conditional that uses msvcrt flags on Python 3.  The problem is that win32file.LockFileEx is being used so the flags don't match, and a non-blocking flag will block instead.  Just remove the "if sys.version" and always use the win32con flags since msvcrt isn't used for SHARED.

**Comment by WoLpH** [comment]
Sorry for the really slow response, I didn't get an email about this issue. 

I'm honestly a bit hesitant to change this behaviour at this point for 2 reasons:
1. I can't easily test the possible scenarios
2. It breaks backwards compatibility and some people might expect these results.

It does sound like a serious bug though... but I'm wondering if the better fix isn't to replace the win32file locking with msvcrt instead and remove win32file completely.

**Comment by barry-gallagher** [comment]
Unfortunately for me I ran into a Windows limit on the number of locked
files and moved to a different solution.  I understand your hesitance and
leave it to your discretion as to how/if to change the code.

**Comment by martin-eder-zeiss** [comment]
@WoLpH 

Are there any updates about this? The shared lock feature is completely broken for me right now.

**Comment by WoLpH** [comment]
I've done some testing in the mean time and I believe the proposed solution is probably the best one. Even though it might break backwards compatibility in a few select cases.

I'll create a new release within the next few days

**Comment by WoLpH** [comment]
@martin-eder-zeiss Can you test the current develop branch? I think I've fixed it but I'm somewhat worried I might have broken something...

It seems to work flawlessly for me on a Windows machine though, but I might still be missing other test cases

**Comment by martin-eder-zeiss** [comment]
I am not familiar with Windows file locking mechanisms, but is it intended that this does not throw an AlreadyLocked exception?
```
portalocker.lock(file, portalocker.constants.LOCK_EX)
portalocker.lock(file, portalocker.constants.LOCK_SH | portalocke_.constants.LOCK_NB)
```

How can I get a shared lock when there is already an exclusive one?

**Comment by WoLpH** [comment]
I tested it like this, and that seems to work as expected:
```python
import portalocker


with portalocker.Lock('x', flags=portalocker.LOCK_EX):
    print('lock A')
    with portalocker.Lock('x', flags=portalocker.LOCK_SH | portalocker.LOCK_NB):
        print('lock B')
```
And I tried this as well:
```python
import portalocker

with open('x', 'w') as a:
    portalocker.lock(a, portalocker.LOCK_EX)
    print('lock A')

    with open('x', 'w') as b:
        portalocker.lock(b, portalocker.LOCK_SH | portalocker.LOCK_NB)
        print('lock B')
```

Could it be that you reused the same `file` handle for both? So something like this:
```python
import portalocker

with open('x', 'w') as a:
    portalocker.lock(a, portalocker.LOCK_EX)
    print('lock A')

    portalocker.lock(a, portalocker.LOCK_SH | portalocker.LOCK_NB)
    print('lock B')
```

**Comment by martin-eder-zeiss** [comment]
yep, I reused the file handle. Nevermind, everything seems to work now.

**Comment by WoLpH** [comment]
> Hello @WoLpH. When the fix will be released?

I've released version 2.5.0 about 3 weeks ago. Is it still broken for you?

**Comment by egorvdot** [comment]
> > Hello @WoLpH. When the fix will be released?
> 
> I've released version 2.5.0 about 3 weeks ago. Is it still broken for you?

Sorry, I got confused in my venv-s. I see updates in latest release. Thanks!
